### PR TITLE
[ML][Pipelines] feat: use origin name and version in request body for all anonymous components

### PR DESCRIPTION
**Pull request by elliotzh**
# Description

1. avoid updating attribute `name` of an anonymous component instance on creation
2. use the origin component name in the body of the creation requests
3. keep previous behavior that anonymous component will be created with a fixed name and a hash version

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by wangchao1230 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py [hunk 1]
```diff
@@ -1480,6 +1478,11 @@ def pipeline_distribution_components(job_in_number, job_in_path):
         created_job: PipelineJob = client.jobs.create_or_update(
             pipeline, experiment_name=experiment_name, continue_on_step_failure=True
         )
+        # Theoretically, we should keep the invalid name in request body,
+        # as component name valid in azureml-components maybe invalid in azure-ai-ml.
+        # So we leave this validation to server-side for now.
+        assert mpi_func._to_rest_object().properties.component_spec["name"] == invalid_component_name
```
Comment (elliotzh) [inline sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py:1484]: @wangchao1230 please help confirm this. Another solution is to automatically correct the invalid name on client-side and maybe give a warning

Comment (wangchao1230) [reply sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py:1484]: can we raise exception for v2 components in client side when name is invalid?

Comment (elliotzh) [reply sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py:1484]: Of course. The problem is (what we tested here), if a component is anonymous, previously we won't raise exception for an invalid name as the name will be overwritten. So raising exception directly may lead to potential regression.

Comment (wangchao1230) [reply sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py:1484]: Let's keep current behavior.

**Pull request merged**